packages/chrome-offscreen-worker/src/messages/root-control.ts

```diff
@@ -0,0 +1,57 @@
+/* These messages appear only in the single offscreen root control channel. */
+
+export type WorkerConstructorParamsPrimitive =
+  Required<ConstructorParameters<typeof Worker>> extends [infer S, infer O]
+    ? [S extends string ? S : never, { [k in keyof O]: O[k] & string }]
+    : never;
```

Comment:
mostly this just excludes `URL` from the worker path url parameter, but it also types the `WorkerOptions` structure as primitives to indicate jsonifiability